src/Orleans/Utils/DictionaryExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Orleans
+{
+    static class DictionaryExtensions
+    {
+        public static TResult Find<TKey, TResult>(this IDictionary<TKey, TResult> dictionary, TKey key) where TResult : class
```

Comment:
I never comment internal code. I believe that all code should be as self-explanatory as possible. Public API - yes, that should be commented even if it's obvious